cpp/src/arrow/python/common.h

```diff
@@ -98,6 +101,10 @@ class ARROW_EXPORT OwnedRef {
 // (e.g. if it is released in the middle of a function for performance reasons)
 class ARROW_EXPORT OwnedRefNoGIL : public OwnedRef {
  public:
+  OwnedRefNoGIL() : OwnedRef() {}
+  OwnedRefNoGIL(OwnedRefNoGIL&& other) : OwnedRef(other.detach()) {}
+  explicit OwnedRefNoGIL(PyObject* obj) : OwnedRef(obj) {}
```

Comment:
Can you use `using OwnedRef::OwnedRef` here?

Reply:
That's what I originally did, but it looks like MSVC doesn't inherit all constructors with it:
https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/build/1.0.5355/job/xy8sve5l9fkyd8qg
(and related commit: https://github.com/apache/arrow/commit/12b680cc8c712a33e2e0e644745a897fc73e6344 )

Reply:
Got it, thanks!